internal/flybarman/node.go

```diff
@@ -140,7 +140,7 @@ wal_retention_policy = main
 	}
 
 	if _, err := os.Stat(n.BarmanCronFile); os.IsNotExist(err) {
-		barmanCronFileContent := `* * * * * /usr/bin/barman cron
+		barmanCronFileContent := `* * * * * . $HOME/.profile; /usr/bin/barman cron >> /proc/1/fd/1 2>/proc/1/fd/2
```

Comment:
Would have to test this to confirm, but now that cron is wrapped within a supervisor process we might not actually need to pipe the logs to the main process.

Reply:
I just spun up a debian:12 docker image, installed cron, added the following crontab entry, and ran `cron -f`:
```
* * * * * root echo test
```

Nothing appeared on stdout, so I think the redirects are still needed.

Reply:
Aight, so this should do the trick.  Basically, you could add this to the bin directory in this project and then modify the file the cronjob calls to `/usr/bin/path-to-barman-cron-file.sh`.  

This will identify the cron supervisor process id, run the `barman cron` command and then log the result. 

```
#!/bin/bash

# Identify the process id of the cron supervisor process
cron_pid=$(pgrep -f '/usr/sbin/cron')
if [ -z "$cron_pid" ]; then
    # Send the error message to main process's stderr
    echo "Failed to resolve cron process id" >> /proc/1/fd/2
    exit 1
fi

barman_output=$(barman cron 2>&1)
barman_exit_code=$?

# Log the result of barman cron
if [ $barman_exit_code -ne 0 ]; then
  echo "Barman cron failed with exit code $barman_exit_code: $barman_output" >> /proc/$cron_pid/fd/2
  exit 1
else
  echo "Barman cron succeeded: $barman_output" >> /proc/$cron_pid/fd/1
fi
```

Reply:
Thanks for doing this legwork. My launchpad issues cleared up, so I was finally able to build this and put it through its paces as well as I could.

I've pushed what's on this branch to [nolan/postgres-flex](https://hub.docker.com/r/nolan/postgres-flex) to make testing a bit easier. Unfortunately we don't seem to support `fly pg barman create --image-ref ...` for easier testing, unless I'm missing something. If no one else gets to that, I can do it on Monday.

Reply:
Yeah, this is annoying.  You have to manually update the machine:

```
fly machines <machine-id> --image <image-ref>
```